### PR TITLE
docs: add URL to Log entry deletion page

### PR DESCRIPTION
**Pull request by sarah-spang**
Current doc currently says the API to delete logs is still under development, however, this information appears outdated as there is a `Log entry deletion` doc that provides details on how to enable this now.

**What this PR does / why we need it**:
Remove outdated information and add details and link to the **Log entry deletion** doc.

### CODE REVIEW

File: docs/sources/operations/storage/retention.md [hunk 1]
```diff
@@ -186,19 +186,16 @@ policy set correctly. For more details check
 or
 [GCS's documentation](https://cloud.google.com/storage/docs/managing-lifecycles).
 
-Currently, the retention policy can only be set globally. A per-tenant retention
-policy with an API to delete ingested logs is still under development.
-
-Since a design goal of Loki is to make storing logs cheap, a volume-based
-deletion API is deprioritized. Until this feature is released, if you suddenly
-must delete ingested logs, you can delete old chunks in your object store. Note,
+If you suddenly must delete ingested logs, you can delete old chunks in your object store. Note,
```
Comment (JStickler) [inline docs/sources/operations/storage/retention.md:189]: "Suddenly" just sounds odd here.
```suggestion
If you must delete ingested logs, you can delete old chunks in your object store. Note,
```

File: docs/sources/operations/storage/retention.md [hunk 2]
```diff
@@ -186,19 +186,16 @@ policy set correctly. For more details check
...
 however, that this only deletes the log content and keeps the label index
 intact; you will still be able to see related labels but will be unable to
 retrieve the deleted log content.
 
 For further details on the Table Manager internals, refer to the
 [Table Manager]({{<relref "table-manager">}}) documentation.
 
+Alternatively, if the BoltDB Shipper is configured for the index store, you can enable [Log entry deletion]({{<relref "logs-deletion">}}) to delete log entries from a specific stream.
```
Comment (JStickler) [inline docs/sources/operations/storage/retention.md:197]:
```suggestion
Alternatively, if the BoltDB Shipper is configured for the index store, you can enable [Log entry deletion]({{<relref "logs-deletion.md">}}) to delete log entries from a specific stream.
```

**Review by JStickler — COMMENTED** [review]
[docs team]  Someone from engineering will need to confirm the status of the API.

**Comment by GrafanaWriter** [comment]
Moving to Blocked from a docs perspective as we can't move forward until we have a technical review.

**Comment by GrafanaWriter** [comment]
@JStickler - should we close this as no one has looked at in quite some time?

File: docs/sources/operations/storage/retention.md [hunk 3]
```diff
@@ -211,20 +211,16 @@ policy set correctly. For more details check
...
-must delete ingested logs, you can delete old chunks in your object store. Note,
+If you must delete ingested logs, you can delete old chunks in your object store. Note,
 however, that this only deletes the log content and keeps the label index
 intact; you will still be able to see related labels but will be unable to
 retrieve the deleted log content.
 
 For further details on the Table Manager internals, refer to the
 [Table Manager](https://grafana.com/docs/loki/<LOKI_VERSION>/operations/storage/table-manager/) documentation.
 
+Alternatively, if the BoltDB Shipper is configured for the index store, you can enable [Log entry deletion]({{<relref "logs-deletion.md">}}) to delete log entries from a specific stream.
```
Comment (JStickler) [inline docs/sources/operations/storage/retention.md:222]:
```suggestion
Alternatively, if the BoltDB Shipper is configured for the index store, you can enable [Log entry deletion](https://grafana.com/docs/loki/<LOKI_VERSION>/operations/storage/logs-deletion/) to delete log entries from a specific stream.
```

**Review by JStickler — APPROVED** [review]

**Pull request merged**